### PR TITLE
Migration script enhancement for anonymours users

### PR DESCRIPTION
**Pull request by utk14**

### CODE REVIEW

File: etl-jobs/src/main/scala/org/sunbird/analytics/jobs/UserCacheIndexer.scala [hunk 1]
```diff
@@ -284,11 +289,38 @@ object UserCacheIndexer extends Serializable {
...
 
-    val res1 = time(populateUserData())
-    val res2 = time(denormUserData())
-    val totalTimeTaken = (res1._1 + res2._1).toDouble/1000
-    Console.println("Time taken for individual steps:", "stage1", res1._1, "stage2", res2._1)
-    Console.println("Time taken for complete script:", totalTimeTaken);
+    if (!populateAnonymousData.equalsIgnoreCase("true")) {
+      val res1 = time(populateUserData())
+      val res2 = time(denormUserData())
+      val totalTimeTaken = (res1._1 + res2._1).toDouble/1000
```
Comment (SanthoshVasabhaktula) [inline etl-jobs/src/main/scala/org/sunbird/analytics/jobs/UserCacheIndexer.scala:295]: Redis db is incorrect for registered users

File: etl-jobs/src/main/scala/org/sunbird/analytics/jobs/UserCacheIndexer.scala [hunk 2]
```diff
@@ -284,11 +289,38 @@ object UserCacheIndexer extends Serializable {
...
+      Console.println("Time taken for individual steps:", "stage1", res1._1, "stage2", res2._1)
+      Console.println("Time taken for complete script:", totalTimeTaken);
+    } else {
+      val sqlContext = new SQLContext(spark.sparkContext)
+      import sqlContext.implicits._
+
+      val anonymousDataRDD = spark.sparkContext.fromRedisKV("*").persist(StorageLevel.MEMORY_ONLY)
+      val filteredData = anonymousDataRDD.map{f => (f._1, JSONUtils.deserialize[Map[String, AnyRef]](f._2))}.filter(f => f._2.getOrElse("usersignintype", "").equals("Anonymous"))
```
Comment (SanthoshVasabhaktula) [inline etl-jobs/src/main/scala/org/sunbird/analytics/jobs/UserCacheIndexer.scala:303]: No need to persist the anonymous data frame

Comment (utk14) [reply etl-jobs/src/main/scala/org/sunbird/analytics/jobs/UserCacheIndexer.scala:303]: Done

File: etl-jobs/src/main/scala/org/sunbird/analytics/jobs/UserCacheIndexer.scala (hunk 1 above)
Comment (utk14) [reply etl-jobs/src/main/scala/org/sunbird/analytics/jobs/UserCacheIndexer.scala:296]: Done

**Review by SanthoshVasabhaktula — APPROVED** [review]

**Pull request merged**